### ISSUE TITLE
Accepting DelayedFormat objects everywhere we accept Paths?

### ISSUE THREAD

**Issue by curufinwe** [rwth-i6/i6_core#198]
We at Apptek have a use-case where we'd like to pass a DelayedFormat object into the PipelineJob, but that one only accepts Paths. The DelayedFormat is used to append a static relative path to the output path of a CloneGitRepositoryJob. I can think of different ways to solve this:
1. Change the recipes to call `get()` instead of `get_path()` everywhere.
2. Change sisyphus to support creating a Path from a DelayedFormat object

The first option is nice because it's fairly simple to do, whereas the second option needs quite a bit of changes inside the Path class (but maybe it only looks to me that way as I don't know the code very well). Maybe @critias can comment on that.
The first option has the side-effect that we now accept also other stuff that is not necessarily path in many places, i.e. a Variable. It's not really a deal-breaker as the code would have also crashed before, now it would just crash in a different way (while trying to open a non-existing file vs. calling a non-existing method).

@JackTemaki @michelwi @mmz33 what do you think?

**Comment by critias** [comment]
Problem with proposal 1. get_cached_path will not work any more
Problem with proposal 2. it's not possible to create a Path object from every DelayedFormat object

Proposal  3.:
Change Sisyphus to create a Path from a Path object instead of a DelayedFormat object. Basically overwriting __add__ and __get__ to return a Path object with an, according to the operation, updated local path.

This should solve your problem, right? Or am I missing anything?

**Comment by michelwi** [comment]
I don't think you need a delayed operation here. At the time of graph creation the output_path is already known as well as the static addition at the end.

You should be able to create a (non-delayed) Path object to your target directly.

**Comment by michelwi** [comment]
e.g.

```
new_path = Path(path=old_path.path+addition, creator=old_path.creator, ...)
```

Which is basically @critias Proposal 3 executed manually.

**Comment by curufinwe** [comment]
Yeah, I like the idea of adding some support for that in sisyphus to make it less cumbersome. I'll keep the issue open for now (until we update sisyphus).

**Comment by JackTemaki** [comment]
fixed with #71
